packages/client/src/views/SysAdmin/Config/Application/DynamicModal.tsx

```diff
@@ -183,434 +180,308 @@ interface IProps {
     notificationStatus: NOTIFICATION_TYPE,
     messages: string
   ) => void
-  offlineCountryConfiguration: IOfflineData
-}
-
-type DispatchProps = {
-  updateConfig: typeof updateOfflineConfigData
 }
 
-export type IFullProps = IProps & IntlShapeProps & DispatchProps
-class DynamicModalComponent extends React.Component<IFullProps, IState> {
-  constructor(props: IFullProps) {
-    super(props)
-    const { offlineCountryConfiguration } = this.props
-    this.state = {
-      applicationName: offlineCountryConfiguration.config.APPLICATION_NAME,
-      currency: `${offlineCountryConfiguration.config.CURRENCY.languagesAndCountry[0]}-${this.props.offlineCountryConfiguration.config.CURRENCY.isoCode}`,
-      birthRegistrationTarget:
-        offlineCountryConfiguration.config.BIRTH.REGISTRATION_TARGET.toString(),
-      birthLateRegistrationTarget:
-        offlineCountryConfiguration.config.BIRTH.LATE_REGISTRATION_TARGET.toString(),
-      deathRegistrationTarget:
-        offlineCountryConfiguration.config.DEATH.REGISTRATION_TARGET.toString(),
-      birthOnTimeFee:
-        offlineCountryConfiguration.config.BIRTH.FEE.ON_TIME.toLocaleString(),
-      birthLateFee:
-        offlineCountryConfiguration.config.BIRTH.FEE.LATE.toLocaleString(),
-      birthDelayedFee:
-        offlineCountryConfiguration.config.BIRTH.FEE.DELAYED.toLocaleString(),
-      deathOnTimeFee:
-        offlineCountryConfiguration.config.DEATH.FEE.ON_TIME.toLocaleString(),
-      deathDelayedFee:
-        offlineCountryConfiguration.config.DEATH.FEE.DELAYED.toLocaleString(),
-      nidPattern:
-        props.offlineCountryConfiguration.config.NID_NUMBER_PATTERN.toString(),
-      nidExample: EMPTY_STRING,
-      testNid: false,
-      phoneNumberPattern:
-        props.offlineCountryConfiguration.config.PHONE_NUMBER_PATTERN.toString(),
-      phoneNumberExample: EMPTY_STRING,
-      testPhoneNumber: false,
-      updatingValue: false,
-      errorOccured: false,
-      errorMessages: EMPTY_STRING,
-      govtLogo: EMPTY_STRING,
-      logoFile: { name: EMPTY_STRING, type: EMPTY_STRING, data: EMPTY_STRING },
-      isFileUploading: false,
-      logoFileName: EMPTY_STRING
-    }
-  }
-
-  showChangeModal: boolean = !!!this.props.changeModalName ? false : true
-
-  setApplicationName = (event: React.ChangeEvent<HTMLInputElement>) => {
+export type IFullProps = IProps
+
+function DynamicModalComponent(props: IProps) {
```

Comment:
It would be a good idea to break this component into smaller ones perhaps like _GeneralActionModal_, _BirthActionModal_ and _DeathActionModal_. But for now, I reckon this is a good start.